jvb/src/main/kotlin/org/jitsi/videobridge/Endpoint.kt

```diff
@@ -1172,6 +1088,79 @@ class Endpoint @JvmOverloads constructor(
         private val random = SecureRandom()
     }
 
+    private inner class SctpCallbacks(transport: DcSctpTransport) : DcSctpBaseCallbacks(transport) {
+        override fun sendPacketWithStatus(packet: ByteArray): SendPacketStatus {
+            val newBuf = ByteBufferPool.getBuffer(packet.size)
+            System.arraycopy(packet, 0, newBuf, 0, packet.size)
+
+            sctpSendPcap.observe(newBuf, 0, packet.size)
+            dtlsTransport.sendDtlsData(newBuf, 0, packet.size)
+
+            return SendPacketStatus.kSuccess
+        }
+
+        override fun OnMessageReceived(message: DcSctpMessage) {
+            // We assume all data coming over SCTP will be datachannel data
+            val dataChannelPacket = DataChannelPacket(message)
+            // Post the rest of the task here because the current context is
+            // holding a lock inside the SctpSocket which can cause a deadlock
+            // if two endpoints are trying to send datachannel messages to one
+            // another (with stats broadcasting it can happen often)
+            incomingDataChannelMessagesQueue.add(PacketInfo(dataChannelPacket))
+        }
+
+        override fun OnError(error: ErrorKind, message: String) {
+            logger.warn("SCTP error $error: $message")
+        }
+
+        override fun OnAborted(error: ErrorKind, message: String) {
+            logger.warn("SCTP aborted with error $error: $message")
+        }
+
+        override fun OnConnected() {
+            logger.info("SCTP connection is ready, creating the Data channel stack")
+            dataChannelStack = DataChannelStack(
+                { data, sid, ppid ->
+                    val message = DcSctpMessage(sid.toShort(), ppid, data.array())
+                    val status = sctpTransport?.socket?.send(message, DcSctpTransport.DEFAULT_SEND_OPTIONS)
+                    return@DataChannelStack if (status == SendStatus.kSuccess) {
+                        0
+                    } else {
+                        logger.error("Error sending to SCTP: $status")
+                        -1
+                    }
+                },
+                logger
+            )
+            // This handles if the remote side will be opening the data channel
+            dataChannelStack!!.onDataChannelStackEvents { dataChannel ->
```

Comment:
One thing I've done in situations like this is assign the newly-created thing to a local variable, so then the compiler knows it's non-null.  You can assign it to the member somewhere (like a `this.dataChannelStack = dataChannelStack` at the end or right after it's created if there are early-return scenarios), but use it via the local reference then you don't need all the `!!` in the calls.